Add docs to lc proxy (#1248)
status-im/nimbus-eth1: lc_proxy/docs/metamask_configuration.md

> # Metamask configuration with Alchemy provider

Reviewer: ```suggestion
# MetaMask configuration with Alchemy provider
```

> This documents shows how to bulild and start light client proxy on goerli
> network and pair it with metamask browser extension

Reviewer: ```suggestion
This documents shows how to build and start light client proxy on Goerli
network and pair it with the MetaMask browser extension.
```

### 1. Build light client proxy

First build light client proxy as explained [here](../README.md#Build-light-client-proxy).

### 2. Configuring and running light client proxy

> To run binary built in prvious step with goerli config and using alchemy data
> provider run:

Reviewer: ```suggestion
To run the binary built in previous step with Goerli config and using Alchemy data
provider, run:
```

```bash
# From the nimbus-eth1 repository

> ./build/lc_proxy --trusted-block-root:TrustedBloockRoot --web3-url="wss://eth-goerli.g.alchemy.com/v2/ApiKey" --network=goerli

Reviewer: ```suggestion
./build/lc_proxy --trusted-block-root:TrustedBlockRoot --web3-url="wss://eth-goerli.g.alchemy.com/v2/ApiKey" --network=goerli
```

```

> `ApiKey` - needs to be personal key assigned by alchemy

Reviewer: ```suggestion
`ApiKey`: personal API key assigned by Alchemy
```

> `TrustedBloockRoot` - need to be trusted block root, from which light client will
> start synchronization.

Reviewer: ```suggestion
`TrustedBlockRoot`: Trusted block root, from which the consensus light client will
start synchronization
```

> This command also starts http server with address `http://127.0.0.1:8545` to listen
> for incoming json rpc request.

Reviewer: ```suggestion
This command also starts an HTTP server with address `http://127.0.0.1:8545` to listen
for incoming JSON RPC requests.
```

> After startup, light client will start looking for suitable peers in the network
> i.e peers which serves light client data and then start syncing with the network.

Reviewer: ```suggestion
After startup, light client will start looking for suitable peers in the network,
i.e peers which serve light client data, and will then start syncing.
```

> Until light client syncs with the network, most of the rpc endpoints will be inactive
> and will fail to respond to queries. This happens because until light client syncs up
> with the network, light client proxy can't verify responses from data provider.

Reviewer: ```suggestion
During syncing most of the RPC endpoints will be inactive
and will fail to respond to queries. This happens the light client proxy can't verify responses
from the data provider until the consensus light client is in sync with the consensus chain.
```

> When light client sync up with the network following line should be visible in the

Reviewer: ```suggestion
When the light client is in sync, the following line should be visible in the
```

logs:

```bash
NOT 2022-09-29 10:06:15.974+02:00 New LC optimistic block                    opt=81de61ec:3994230 wallSlot=3994231
```

> After receiving first optimistic block, proxy is ready to be used with metamask

Reviewer: ```suggestion
After receiving the first optimistic block, the proxy is ready to be used with MetaMask.
```

> ### 3. Configuring metamask extension to use custom network

Reviewer: ```suggestion
### 3. Configuring MetaMask extension to use custom network
```

To add custom network in metamask browser extension:
1. Go to `settings`
2. In `settings`, go to `networks` tab
3. There should be `Add a network` button.

> 4. Most important fields when adding new network are `New RPC URL` and `Chain ID`

Reviewer: ```suggestion
4. The most important fields when adding a new network are `New RPC URL` and `Chain ID`.
```

`New RPC URL` should be configured to point to http server started with proxy, in this
example it will be `http://127.0.0.1:8545`. `Chain ID` should be set to chain if of
the network used by proxy, so for goerli it will be equal to `5`

If everyting went smooth there should be new network in `Networks` drop down menu.

After switching to this network it should be possible to create new accounts, and
perform transfers between them

NOTE: Currently when adding custom network with the chain id which is already existis in metamask
configuration, metamask will highlight this as an error. This should be ignored

> as this is really a warning, and it is known bug in metamask.

Author: added link to the bug
